docs/sessions.md

```diff
@@ -0,0 +1,127 @@
+---
+layout: default
+---
+
+State and Sessions in Suave
+===========================
+
+While stateless web applications are all the rage, there are still times where state needs to be preserved, either across requests in a session, or simply among `WebPart`s as the request/response pipeline is composed.  In these cases, Suave has you covered!  We'll look at three aspects of session cookies first, then how we can construct state among `WebPart`s.
+
+Cookie State Session Storage
+----------------------------
+
+Suave provides storage interfaces for cookies in the `Suave.State.CookieStateStore` module.  The `statefulForSession` WebPart can be composed to make a path session-aware.  From there, the `HttpContext.state` function extracts the state information, and the `get` and `set` functions on the resulting `StateStore` object can be used to manipulate the contents of the state tracked by the session.
+
+{% highlight fsharp %}
+open Suave
+open Suave.State.CookieStateStore
+
+/// string -> 'T -> WebPart
+let setSessionValue key value = context (fun ctx ->
+  match ctx |> HttpContext.state with
+  | Some state -> state.set key value
+  | _ -> never // fail)
+
+/// HttpContext -> string -> 'T option
+let getSessionValue ctx key =
+  match ctx |> HttpContext.state with
+  | Some state -> state.get key
+  | _ -> None
+
+/// HttpContext -> string -> string
+let getStringSessionValue ctx key = 
+  match getSessionValue ctx key with
+  | Some value -> string value
+  | _ -> ""
+
+/// WebPart
+let cookieYes = warbler (fun ctx -> OK (getStringSessionValue ctx "test"))
+
+/// WebPart
+let cookieNo = warbler (fun ctx -> OK (getStringSessionValue ctx "nope"))
+
+/// WebPart
+let app =
+  statefulForSession
+  >=> setSessionValue "test" "123"
+  >=> choose [
+        path "/yes" >=> cookieYes
+        path "/no" >=> cookieNo
+        RequestErrors.NOT_FOUND
+        ]
+{% endhighlight %}
+
+Server Keys
+-----------
+
+The contents of the cookie are encrypted before the cookie is sent. Suave's default configuration generates a new server key each time the server is restarted. While this is not _wrong_, users would likely get quite annoyed if they lost their state because the server was restarted. Additionally, specifying a server key lets load-balanced servers access the same information.
```

Comment:
Well it's not wrong, that's true ;), it's to make the developer understand they need to **change** the key; otherwise **all** Suave apps would be running the same key and that would be a security vulnerability. Perhaps add that to the description?